cloud/aws/actuators/machine/actuator.go

```diff
@@ -498,6 +499,14 @@ func (a *Actuator) updateStatus(machine *clusterv1.Machine, instance *ec2.Instan
 	} else {
 		mLog.Debug("status unchanged")
 	}
+
+	// If machine state is still pending, we will return an error to keep the controllers
+	// attempting to update status until it hits a more permanent state. This will ensure
+	// we get a public IP populated more quickly.
+	if awsStatus.InstanceState != nil && *awsStatus.InstanceState == ec2.InstanceStateNamePending {
+		mLog.Infof("instance state still pending, returning an error to requeue")
+		return &machinecontroller.RequeueAfterError{RequeueAfter: 20 * time.Second}
```

Comment:
may be move 20 to a const

Reply:
Will do in the upstream PR and here as well. Thanks.